homework6/src/main/java/Main.java

```diff
@@ -0,0 +1,300 @@
+
+//    4) Generate 5 random Strings and add them to a Set. Find the "max" (while explaining as Javadoc how comparing Strings works)
+        Set<String> randomStringsSet = generateRandomStringSet();
+        System.out.println("\nStrings in the set are: ");
+        randomStringsSet.forEach(System.out::println);
+        System.out.println("\nThe last String in alphabetical order: " + maxFromStrings(randomStringsSet));
+
+//    5) Generate a random number of Integers and then count them.
+//       "Map" the exponential to the numbers and then print them out.
+        randomNumbersInMap(randomNumbersInList());
+
+//    6) Create a map of "n" (K,V) elements and print "how many" elements have value over 10
+//       (the key is of your choice)
+        nElementInMap(15);
+
+//    7) Sort the above Set<String> (used for max) in reverse order
+        System.out.println("\nThe content of the Set: ");
+        randomStringsSet.forEach(System.out::println);
+        System.out.println("\nThe above Set<String> in reverse order: ");
+        sortingStringsInReverseOrder(randomStringsSet).forEach(System.out::println);
+
+
+//    8) Sort the above List of custom objects (used for filter) in an order you consider
+        System.out.println("\nThe sorted Set in reverse order: ");
+        sortingCustomObjects(carsWithM).forEach(System.out::println);
+
+//    9) Check if any of your instances "match" a condition based on an Integer field (of your choice).
+//       What does it return ? Print it out.
+        if (checkIf(carList)) System.out.println("\nAt least one car has a power greater than 250.");
+        else System.out.println("\nNo result found");
+
+//    10) What does Optional represent ? Explain through an example on your custom object
+//          -> Wrap an existing instance
+//          -> Wrap a null
+//          -> Check value using ifPresent or isPresent
+        Car car6 = new Car("Mercedes", "CLS", Flue.DIESEL_HYBRID, 250, 2.0, 2019, Optional.of("white"));
+        carList.add(car6);
+        Car car7 = new Car("Maserati", "Granturismo", Flue.PETROL, 405, 4.3, 2009, Optional.ofNullable("null"));
+        carList.add(car7);
+
+        System.out.println("\nColors specified at some cars: ");
+        colorIsPresent(carList).forEach(System.out::println);
+
+//     11) Fastest way to find the shortest String in a List
+//         (take the 5 random Strings generated above and find the shortest one).
+        System.out.println("\nElements of the set: ");
+        randomStringsSet.forEach(System.out::println);
+        System.out.println("\nShortest String: " + shortestString(randomStringsSet));
+
+    }
+
+    /**
+     * this method finds the shortest String in the Set
+     *
+     * @param set Set of String
+     * @return String type
+     */
+    private static String shortestString(Set<String> set) {
+        return set.stream().min(Comparator.comparingInt(String::length)).get();
```

Comment:
it's not actually the fastest way, it depends on how many elements there are in the list (reducer ?)

Reply:
what happens if they are the same in length ?